plugins/flatpak/gs-flatpak.c

```diff
@@ -1348,12 +1401,24 @@ gs_flatpak_add_updates (GsFlatpak *self, GsAppList *list,
 			g_warning ("failed to add flatpak: %s", error_local->message);
 			continue;
 		}
-		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
-		gs_app_set_update_details (app, NULL);
-		gs_app_set_update_version (app, NULL);
-		gs_app_set_update_urgency (app, AS_URGENCY_KIND_UNKNOWN);
-		gs_app_set_size_download (app, 0);
-		gs_app_list_add (list, app);
+
+		if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME)
+			main_app = get_main_app_of_related (self, app, cancellable,
+							    &error_local);
+
+		if (main_app == NULL)
+			main_app = app;
```

Comment:
Maybe print `error_local->message` with a `g_warning()` here would be good.

Reply:
If `main_app` gains e.g. autoptr, you'll need to increase the ref here.